### PR TITLE
feat(gas-oracle): add gas price update after Curie

### PR DESCRIPTION
**Pull request by colinlyguo**
### Purpose or design rationale of this PR

This PR adds gas price update after curie and fixes some missing unit tests.

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [x] feat: A new feature

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [x] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change

### CODE REVIEW

File: rollup/internal/controller/relayer/l1_relayer_metrics.go [hunk 1]
```diff
@@ -9,7 +9,8 @@ import (
 
 type l1RelayerMetrics struct {
 	rollupL1RelayerGasPriceOraclerRunTotal      prometheus.Counter
-	rollupL1RelayerLastGasPrice                 prometheus.Gauge
+	rollupL1RelayerLastestBaseFee               prometheus.Gauge
```
Comment (Thegaram) [inline rollup/internal/controller/relayer/l1_relayer_metrics.go]:
```suggestion
	rollupL1RelayerLatestBaseFee               prometheus.Gauge
```

File: rollup/internal/controller/relayer/l1_relayer.go [hunk 2]
```diff
@@ -219,3 +231,29 @@ func (r *Layer1Relayer) StopSenders() {
...
+
+	expectedBaseFeeDelta := r.lastBaseFee*r.gasPriceDiff/gasPriceDiffPrecision + 1
+	if baseFee >= r.minGasPrice && (baseFee >= r.lastBaseFee+expectedBaseFeeDelta || baseFee+expectedBaseFeeDelta <= r.lastBaseFee) {
+		return true
+	}
+
+	// Omitting blob base fee checks before Curie.
+	if !isCurie {
+		return false
+	}
```
Comment (Thegaram) [inline rollup/internal/controller/relayer/l1_relayer.go:249]:
```suggestion
	}
	
	if r.lastBlobBaseFee == 0 {
		return true
	}
```

Comment (colinlyguo) [reply rollup/internal/controller/relayer/l1_relayer.go:249]: added in [23f46af](https://github.com/scroll-tech/scroll/pull/1344/commits/23f46afcb55dae3d8b317dec27d537b9fbd61f33).

File: rollup/internal/controller/relayer/l1_relayer_metrics.go (hunk 1 above)
Comment (colinlyguo) [reply rollup/internal/controller/relayer/l1_relayer_metrics.go]: fixed in [23f46af](https://github.com/scroll-tech/scroll/pull/1344/commits/23f46afcb55dae3d8b317dec27d537b9fbd61f33).

**Review by zimpha — DISMISSED** [review]

File: rollup/internal/controller/relayer/l1_relayer.go [hunk 3]
```diff
@@ -169,9 +178,12 @@ func (r *Layer1Relayer) ProcessGasPriceOracle() {
...
 				return
 			}
-			r.lastGasPrice = baseFee
-			r.metrics.rollupL1RelayerLastGasPrice.Set(float64(r.lastGasPrice))
-			log.Info("Update l1 base fee", "txHash", hash.String(), "baseFee", baseFee, "isBernoulli", isBernoulli)
+
+			r.lastBaseFee = baseFee
+			r.lastBlobBaseFee = blobBaseFee
+			r.metrics.rollupL1RelayerLatestBaseFee.Set(float64(r.lastBaseFee))
+			r.metrics.rollupL1RelayerLastestBlobBaseFee.Set(float64(r.lastBlobBaseFee))
```
Comment (Thegaram) [inline rollup/internal/controller/relayer/l1_relayer.go]:
```suggestion
			r.metrics.rollupL1RelayerLatestBlobBaseFee.Set(float64(r.lastBlobBaseFee))
```

Comment (colinlyguo) [reply rollup/internal/controller/relayer/l1_relayer.go]: oh I missed this, fixed in [43ce845](https://github.com/scroll-tech/scroll/pull/1344/commits/43ce84515400a7dbc2ae2c01a6fd71fe961d8dce).

**Review by georgehao — APPROVED** [review]

**Review by 0xmountaintop — APPROVED** [review]

**Pull request merged**